doc/guides/getting_started/src/e2e_regression_setup.md

```diff
@@ -0,0 +1,55 @@
+# ROM E2E Regressions setup
+This guide will help you to setup an environment to run the ROM E2E tests in an Ubuntu 20.04 machine.
+
+## Introduction
+The [ROM](https://opentitan.org/book/doc/security/specs/secure_boot/index.html?highlight=ROM#rom-rom) is first boot stage of secure boot and it is placed in the ROM memory which means that it cannot be updated after manufacturing at all.
+The ROM E2E (End-to-End) tests validates the ROM features and can be used for regression tests.
+Each test is divided into components:
+ - **Opentitantool**, it is a Rust program that provide a common interface to interact with Opentitan. Opentitantool uses opentitanlib which provides the abstractions to communicate with Opentitan.
+ - **Host test** software, it is a Rust program built by bazel that actively interacts with the device software via UART, SPI or JTAG to perform the test and check whether it passed or not. To avoid code duplication the tests uses the `opentitanlib` to communicate with opentitan.
+ - **Device test** software, it is a C program built by bazel that run in the flash after the ROM and perform checking in the HW and/or memories to check the configurations performed by the ROM.
```

Comment:
```suggestion
 - **Device test** software, a C program (built by `bazel` to run on an OpenTitan device) that is loaded into flash, and booted by the ROM stage. It checks the HW and/or memory configurations performed by the ROM and relays information to the Host test software.
```
The tests are loaded into flash but are copied into SRAM before they are run by the C runtime, so technically they run out of SRAM, but are loaded into flash.

Reply:
Interesting, out of curiosity, why we run them from SRAM, performance requirements?